File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithComponentCollisionAndDirect.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithComponentCollisionAndDirect {
    @Root
    default Class1 c1() {
        return new Class1();
    }

    @Root
    default Class1 c2() {
        return new Class1();
    }

    @Root
    default Class1 c3() {
        return new Class1();
    }

    @Root
    default Class2 class2(Class1 class1) {
        return new Class2(class1);
    }


    class Class1 {}

    record Class2(Class1 class1) {}
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithComponentCollisionAndDirect.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithComponentCollisionAndDirect {
    fun c1(): Class1 {
        return Class1()
    }

    fun c2(): Class1 {
        return Class1()
    }

    fun c3(): Class1 {
        return Class1()
    }

    @Root
    fun class2(class1: Class1): Class2 {
        return Class2(class1)
    }

    class Class1
    data class Class2(val class1: Class1)
}
